Code/Framework/AzQtComponents/AzQtComponents/Components/InputDialog.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "InputDialog.h"
+
+#include <QEventLoop>
+#include <QInputDialog>
+#include <QRegularExpressionValidator>
+
+namespace AzQtComponents
+{
+    InputDialog::InputDialog(QWidget* parent)
+        : QInputDialog(parent)
+        , m_validator(nullptr)
+    {
+    }
+
+    void InputDialog::setValidator(QValidator* validator)
+    {
+        m_validator = validator;
+
+        // If this is called after show has been called, the QLineEdit will be available to patch.
+        QLineEdit* lineEdit = this->findChild<QLineEdit*>();
+        if (lineEdit)
+        {
+            lineEdit->setValidator(m_validator);
+        }
```

Comment:
Nit: Could move this logic to a private helper method since we do the same logic on `show()`

Reply:
That's done. I've also removed the assignment being conditional on the validator existing to allow the user to remove a validator if they want to.